src/main/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterLoader.java

```diff
@@ -0,0 +1,56 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import org.reflections.Reflections;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+public class NamePrinterLoader implements Iterable<NamePrinter> {
+
+    //Singleton implementation
+    private static NamePrinterLoader instance;
+    private final String packageIdentifier;
+    private List<NamePrinter> namePrinters;
+    private NamePrinterLoader() {
+        packageIdentifier = getClass().getPackage().getName();
+
+        try {
+            namePrinters = getAllNamePrinters();
+        } catch (Exception e) {
+            e.printStackTrace();
+            namePrinters = null;
+        }
+    }
+
+    public static NamePrinterLoader getInstance() {
+        if (instance == null)
+            instance = new NamePrinterLoader();
+
+        return instance;
+    }
+
+    @Override
+    public Iterator<NamePrinter> iterator() {
+        return namePrinters.iterator();
+    }
+
+    private List<NamePrinter> getAllNamePrinters() throws Exception {
+        List<NamePrinter> namePrinters = new ArrayList<>();
+        Reflections reflections = new Reflections(packageIdentifier);
+        Set<Class<? extends NamePrinter>> allClasses = reflections.getSubTypesOf(NamePrinter.class);
+
+        for (Class classFound : allClasses) {
+            if (NamePrinter.class.isAssignableFrom(classFound)
```

Comment:
refactor this to be readable

Reply:
I recommended to extract logic to make it more readable

Reply:
@champillon  @mahasak I renamed `allClasses` to `allNamePrinterClasses` and `classFound` to `namePrinterClass` and moved the logic checking if the class is instantiable to `boolean isInstantiable(Class<? extends NamePrinter>)`

Reply:
problem is not on the variable name but problem inside "if"
the logic inside "if" is too complex and not well-described